src/Centreon/Domain/Monitoring/CommandLineTrait.php

```diff
@@ -43,6 +43,9 @@ private function buildCommandLineFromConfiguration(
         array $macros,
         string $replacementValue
     ): string {
+        // if the command line contains $$ after a macro (so $$$), delete one of them to match with
+        // the command executed by centreon-engine
+        $configurationCommand = str_replace('$$$', '$$', $configurationCommand);
```

Comment:
I think centreon-engine needs `$$` to escape `$`, so this replacement should be enough to fix the issue
```suggestion
        $configurationCommand = str_replace('$$', '$', $configurationCommand);
```

Reply:
The function _buildCommandLineFromConfiguration_ isn't used to generate the configuration. It's used for the monitoring page in order to display the command line.
A lot of plugins packs contain this kind of option:
`--interface='^$_SERVICEINTERFACENAME$$$'`
Because, you're right, centreon-engine needs $$ to execute $
So I wanted to fix this specific case

Reply:
following our discussion, it's better to replace $$ by $